my_project/extract_df.py

```diff
@@ -306,6 +308,49 @@ def create_df(lst, file_name):
 
     return epw_df, location_info
 
+#convert function
+def temperature(df,name):
+    df[name] = df[name]*1.8+32
```

Comment:
@yunzhu1229 if I understand correctly this function can only convert from C to F but cannot go back. Is this the case? What if the user wants to go back to C? Maybe it is not important to implement this feature but you should discuss this with @stefanoschiavon.

Reply:
In the switch_si_ip function, I have defined the data request method. When the state go back to 'si', it will directly request the data again by the url, so the data will go back to 'si' state and needn't do the conversion. So I think the user can flexibly choose the state, all the data can be converted from 'si' to 'ip', and they can also go back to 'si'.

Reply:
okay, thank you for the explanation. The issue with this approach is that we are using resources inefficiently, this will generate an unnecessary network request to download the data and also it will slow down the application. We should either save the SI unit dataframe locally so it can be recalled if the user goes back to the SI selection or implement functions to convert the data back to SI.

Reply:
See the other comments. Because the df-store can only be output once, so I think it should be output at the switch_si_ip function, so that the data after conversion can be used elsewhere. Therefore, it is needed to create the df in the switch_si_ip function. At the same time, the back convert from PI to SI is no needed, it can be done just by create the df again.